zcash_client_sqlite/src/error.rs

```diff
@@ -166,8 +168,8 @@ impl From<zcash_primitives::memo::Error> for SqliteClientError {
     }
 }
 
-impl From<ShardTreeError<rusqlite::Error>> for SqliteClientError {
-    fn from(e: ShardTreeError<rusqlite::Error>) -> Self {
+impl From<ShardTreeError<Either<io::Error, rusqlite::Error>>> for SqliteClientError {
```

Comment:
The problem is that we don't want (for example) to construct a `ShardTreeError<io::Error>` or a `ShardTreeError<rusqlite::Error>`, and neither do we want to provide direct `From` impls from these types to SqliteClientError because we'd then end up in an ambiguous situation.